scripts/gb_model/get_EU_generator_bid_offer_profile.py

```diff
@@ -0,0 +1,218 @@
+# SPDX-FileCopyrightText: gb-dispatch-model contributors
+#
+# SPDX-License-Identifier: MIT
+
+"""
+Calculate EU marginal generator bids and offer price profiles
+"""
+
+import logging
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import pypsa
+
+from scripts._helpers import configure_logging, set_scenario_config
+from scripts.gb_model._helpers import filter_interconnectors, marginal_costs_bus
+
+logger = logging.getLogger(__name__)
+
+
+def _extract_marginal_price_profiles(network: pypsa.Network):
```

Comment:
You're calling it from the __name__ == "__main__" block so make it a public method